src/coaster.cpp

```diff
@@ -1589,6 +1607,90 @@ bool CoasterInstance::PlaceEntranceOrExit(const XYZPoint16 &pos, const bool entr
 	return true;
 }
 
+/**
+ * Update the intensity statistics with a piece of new information.
+ * @param point Position along the coaster.
+ * @param speed Current speed of a train.
+ * @param vg Current vertical G force.
+ * @param hg Current horizontal G force.
+ */
+void CoasterInstance::SampleStatistics(uint32 point, const bool valid, const int32 speed, const int32 vg, const int32 hg)
+{
+	point /= COASTER_INTENSITY_STATISTICS_SAMPLING_PRECISION;
+	auto it = this->intensity_statistics.find(point);
+	if (it == this->intensity_statistics.end()) {
+		this->intensity_statistics.emplace(std::make_pair(point, CoasterIntensityStatistics{valid, 1, speed, vg, hg}));
+	} else {
+		it->second.valid &= valid;
+		it->second.speed        = (it->second.precision * it->second.speed        + speed) / (it->second.precision + 1);
+		it->second.vertical_g   = (it->second.precision * it->second.vertical_g   + vg   ) / (it->second.precision + 1);
+		it->second.horizontal_g = (it->second.precision * it->second.horizontal_g + hg   ) / (it->second.precision + 1);
+		it->second.precision++;
+	}
+}
+
+void CoasterInstance::RecalculateRatings()
+{
+	uint64 exc = 100;
+	uint64 iny = 100;
+	uint64 nau = 100;
```

Comment:
```suggestion
	uint64 exc = 100; // Excitement * 100
	uint64 iny = 100; // Intensity * 100
	uint64 nau = 100; // Nausea * 100
```
I hope `* 100` is correct.